angular-keycloak/src/app/keycloak.service.ts

```diff
@@ -1,24 +1,28 @@
-import { Injectable } from '@angular/core';
-import { BehaviorSubject, Observable, of } from 'rxjs';
-import { filter, map, take } from 'rxjs/operators';
+import {Injectable, NgZone} from '@angular/core';
+import {BehaviorSubject, Observable, of, Subject} from 'rxjs';
+import {filter, map, mergeMap, switchMap, take, tap} from 'rxjs/operators';
 
 declare var Keycloak: any;
 
 @Injectable({
   providedIn: 'root',
 })
 export class KeycloakService {
+  public isAuthenticated$ = new Subject<boolean>();
   private _keycloak$ = new BehaviorSubject<any>(null);
   private keycloak$ = this._keycloak$.pipe(filter((k) => !!k));
-  private firstInitCallOnly = false;
+  private userData$ = new Subject();
 
-  constructor() {
-    console.log(`KeycloakService - ctor`);
-    if (!this.firstInitCallOnly) {
-      this.firstInitCallOnly = true;
-      console.log(`KeycloakService - initKeycloak()`);
-      this.initKeycloak();
-    }
+  constructor(private zone: NgZone) {
+    this.initKeycloak();
+    this.loadUserData();
+  }
+
+  private loadUserData(): void {
+    this.keycloak$.pipe(
+      take(1),
+      switchMap(k => k.loadUserProfile())
```

Comment:
das war total fies, weil loadUserProfile ein http request macht und so change detection triggert. Das kann in ner endlosschleife enden...